### ISSUE TITLE
Cache runtime routes (ISR)

### ISSUE THREAD

**Issue by jbmoelker** [voorhoede/head-start#198]
All HTML routes get a `cache-control: public, max-age=0, must-revalidate` header from Cloudflare. For our pages statically generated during build this is fine. Cloudflare's CDN and caching mechanism has us covered. For pages that are rendered during runtime the responses will be slower and will hit the underlying services (like DatoCMS) more. We can improve that performance by adding caching headers in an Incremental Static Rendering (ISR) pattern. This blog post explains this pattern using Astro and Cloudflare well: https://logsnag.com/blog/implementing-isr-in-astro

In short, on routes we want to add caching to we can do something like this:
```astro
---
// Set Cache-Control headers for CDN caching
Astro.response.headers.set('Cache-Control', 'public, s-maxage=60, stale-while-revalidate=30');
---
```

And we could add a helper to set that header (as lib or more conveniently on `Astro.locals` as we need access to the `Astro` object), like so: `(Astro.locals.)setCache({ seconds: 60 })`.

**Comment by decrek** [comment]
I like this! I would like to not have the helper since I think this is straightforward enough. Do you see this as something that we add to the docs?

**Comment by jbmoelker** [comment]
Related proposal for Astro core: https://github.com/withastro/roadmap/discussions/1131
